modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java

```diff
@@ -1090,21 +1097,16 @@ private void queueWatchEvent() {
                 updatedEntries.clear();
 
                 break;
-
             case IN_PROGRESS:
-                // Buffer the event while event replay is still in progress.
-                if (eventCache == null) {
-                    eventCache = new ArrayList<>();
-                }
+                UpdatedEntries copy = updatedEntries.transfer();
 
-                eventCache.add(updatedEntries.transfer());
+                var event = new UpdateEntriesEvent(copy.updatedEntries, copy.ts);
 
-                break;
+                addToNotifyWatchProcessorEventsBeforeStartWatches(event);
 
+                break;
             default:
                 notifyWatches();
-
-                break;
```

Comment:
It seems consistent to have `break` even in the `default` branch. I would leave it here